### PR TITLE
Add advanced machine feature options

### PR DESCRIPTION
**Pull request by timstoop**
Allow creation of instance templates that either have nested virtualization enabled or support SMT.

### CODE REVIEW

**Comment by timstoop** [comment]
I have no permission to check why the test is failing.

**Comment by morgante** [comment]
```
estInstanceSimpleModule 2022-02-14T17:35:04Z command.go:179: │ Error: Error creating instance: googleapi: Error 400: The selected machine type (n1-standard-1) does not support SMT settings., badRequest
TestInstanceSimpleModule 2022-02-14T17:35:04Z command.go:179: │ 
TestInstanceSimpleModule 2022-02-14T17:35:04Z command.go:179: │   with module.instance_simple.module.compute_instance.google_compute_instance_from_template.compute_instance[2],
TestInstanceSimpleModule 2022-02-14T17:35:04Z command.go:179: │   on ../../../modules/compute_instance/main.tf line 41, in resource "google_compute_instance_from_template" "compute_instance":
TestInstanceSimpleModule 2022-02-14T17:35:04Z command.go:179: │   41: resource "google_compute_instance_from_template" "compute_instance" {
TestInstanceSimpleModule 2022-02-14T17:35:04Z command.go:179: │ 
TestInstanceSimpleModule 2022-02-14T17:35:04Z command.go:179: ╵
TestInstanceSimpleModule 2022-02-14T17:35:04Z command.go:179: ╷
TestInstanceSimpleModule 2022-02-14T17:35:04Z command.go:179: │ Error: Error creating instance: googleapi: Error 400: The selected machine type (n1-standard-1) does not support SMT settings., badRequest
TestInstanceSimpleModule 2022-02-14T17:35:04Z command.go:179: │ 
TestInstanceSimpleModule 2022-02-14T17:35:04Z command.go:179: │   with module.instance_simple.module.compute_instance.google_compute_instance_from_template.compute_instance[0],
TestInstanceSimpleModule 2022-02-14T17:35:04Z command.go:179: │   on ../../../modules/compute_instance/main.tf line 41, in resource "google_compute_instance_from_template" "compute_instance":
TestInstanceSimpleModule 2022-02-14T17:35:04Z command.go:179: │   41: resource "google_compute_instance_from_template" "compute_instance" {
TestInstanceSimpleModule 2022-02-14T17:35:04Z command.go:179: │ 
TestInstanceSimpleModule 2022-02-14T17:35:04Z command.go:179: ╵
TestInstanceSimpleModule 2022-02-14T17:35:04Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; ╷
│ Error: Error creating instance: googleapi: Error 400: The selected machine type (n1-standard-1) does not support SMT settings., badRequest
│ 
│   with module.instance_simple.module.compute_instance.google_compute_instance_from_template.compute_instance[1],
│   on ../../../modules/compute_instance/main.tf line 41, in resource "google_compute_instance_from_template" "compute_instance":
│   41: resource "google_compute_instance_from_template" "compute_instance" {
│ 
╵
╷
│ Error: Error creating instance: googleapi: Error 400: The selected machine type (n1-standard-1) does not support SMT settings., badRequest
│ 
│   with module.instance_simple.module.compute_instance.google_compute_instance_from_template.compute_instance[3],
│   on ../../../modules/compute_instance/main.tf line 41, in resource "google_compute_instance_from_template" "compute_instance":
│   41: resource "google_compute_instance_from_template" "compute_instance" {
│ 
╵
╷
│ Error: Error creating instance: googleapi: Error 400: The selected machine type (n1-standard-1) does not support SMT settings., badRequest
│ 
│   with module.instance_simple.module.compute_instance.google_compute_instance_from_template.compute_instance[2],
│   on ../../../modules/compute_instance/main.tf line 41, in resource "google_compute_instance_from_template" "compute_instance":
│   41: resource "google_compute_instance_from_template" "compute_instance" {
│ 
╵
╷
│ Error: Error creating instance: googleapi: Error 400: The selected machine type (n1-standard-1) does not support SMT settings., badRequest
│ 
│   with module.instance_simple.module.compute_instance.google_compute_instance_from_template.compute_instance[0],
│   on ../../../modules/compute_instance/main.tf line 41, in resource "google_compute_instance_from_template" "compute_instance":
│   41: resource "google_compute_instance_from_template" "compute_instance" {
│ 
╵}
```

**Comment by timstoop** [comment]
All checks passed, going to remove the unnecessary whiteline at the end of the file in my next push and then leave it for your approval!

File: examples/instance_template/simple/main.tf [hunk 1]
```diff
@@ -32,12 +32,14 @@ locals {
...
-  access_config   = [local.access_config]
+  source                       = "../../../modules/instance_template"
+  project_id                   = var.project_id
+  subnetwork                   = var.subnetwork
+  service_account              = var.service_account
+  name_prefix                  = "simple"
+  tags                         = var.tags
+  labels                       = var.labels
+  access_config                = [local.access_config]
+  enable_nested_virtualization = var.enable_nested_virtualization
```
Comment (morgante) [inline examples/instance_template/simple/main.tf:43]: For future reference, it's encouraged to hard code values like this in examples.

**Pull request merged**